pkg/cmd/join/exec.go

```diff
@@ -330,6 +330,12 @@ func (o *Options) run() error {
 		return err
 	}
 
+	//  apply an empty image pull secret if the image pull secret is not found in the ns.
+	err = helpers.ApplyImagePullSecret(context.TODO(), "open-cluster-management", kubeClient)
```

Comment:
please also consider dry-run, and I am not sure if this should be put into the above `applyKlusterlet` func, is this secret used by the klusterlet-operator? if it is, I think we should put it into the `applyKlusterlet` func, and use the constant `OperatorNamesapce` to replace the "open-cluster-management"

Reply:
There is no image pull secret in the operator deployment or sa. so no need to consider the order.